homeassistant/components/sungrow/const.py

```diff
@@ -0,0 +1,181 @@
+
+
+SENSOR_TYPES: tuple[SungrowSensorEntityDescription, ...] = (
+    SungrowSensorEntityDescription(
+        key="time",
+        name="last update",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value=as_local,
+    ),
+    # '5031 - Total active power'
+    SungrowSensorEntityDescription(
+        key="5031 - Total active power",
+        name="power AC",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5017 - Total DC power'
+    SungrowSensorEntityDescription(
+        key="5017 - Total DC power",
+        name="power DC",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5019 - Phase A voltage'
+    SungrowSensorEntityDescription(
+        key="5019 - Phase A voltage",
+        name="voltage AC1",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5020 - Phase B voltage'
+    SungrowSensorEntityDescription(
+        key="5020 - Phase B voltage",
+        name="voltage AC2",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5021 - Phase C voltage'
+    SungrowSensorEntityDescription(
+        key="5021 - Phase C voltage",
+        name="voltage AC3",
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5022 - Phase A current'
+    SungrowSensorEntityDescription(
+        key="5022 - Phase A current",
+        name="current AC1",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5023 - Phase B current'
+    SungrowSensorEntityDescription(
+        key="5023 - Phase B current",
+        name="current AC2",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5024 - Phase C current'
+    SungrowSensorEntityDescription(
+        key="5024 - Phase C current",
+        name="current AC3",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    # '5003 - Daily power yields'
+    SungrowSensorEntityDescription(
+        key="5003 - Daily power yields",
+        name="yield day",
+        icon="mdi:solar-power",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        value=lambda value: round(value, 3),
```

Comment:
You should probably set the state class to INCREASING here (this enables it to appear in the energy metering features in HA)

Reply:
Is TOTAL_INCREASING something for the daily run time? That is the amount of hours the system was online each day.
Summary values are provided by the "running time total" and "yield total".
These two summaries would be different if the device is not connected from the beginning if I understand it correctly.

Reply:
I am not 100% sure.